### PR TITLE
#2487 IDEA BIM API interface

### PR DESCRIPTION
**Pull request by martinjansa**
Generic IDEA BIM API for the future BIM links implementation.

### CODE REVIEW

File: src/IdeaStatiCa.BimApi/IdeaStatiCa.BimApi.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net45</TargetFramework>
+    <TargetFramework>net48</TargetFramework>
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IdeaStatiCa.BimApi.csproj:3]: can we use netstandard ?

Comment (mpospisil) [reply src/IdeaStatiCa.BimApi/IdeaStatiCa.BimApi.csproj:3]: hmm probably not now . The project depends on OpenModel

**Review by mpospisil — DISMISSED** [review]

**Comment by TekuSP** [comment]
Remove "push" and "pull_request" from Github Action.
This should be resolved via Branch protection rules, not by yaml.
Example of .github folder with actions: [https://github.com/TekuSP/Meteostanice-CSharp/tree/master/.github](https://github.com/TekuSP/Meteostanice-CSharp/tree/master/.github)

Idea for future: Github and Devops has compatible "Actions" between each other from 90% and 10% can be made via "ifs". Maybe merge build verification between them?

File: src/IdeaStatiCa.BimImporter/Geometry.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 namespace IdeaStatiCa.BimImporter
 {
-	internal class Geometry : IGeometry
+	public class Geometry : IGeometry
 	{
 		private static readonly IIdeaLogger _logger = IdeaDiagnostics.GetLogger("ideastatica.bimimporter.geometry");
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimImporter/Geometry.cs]: assembly should not depend on IdeaStatiCa.Diagnostics - use its wrapper IPluginLogger https://github.com/idea-statica/ideastatica-public/blob/main/src/IdeaStatiCa.Plugin/IPluginLogger.cs

**Review by mpospisil — DISMISSED** [review]

**Review by TekuSP — DISMISSED** [review]

File: src/IdeaStatiCa.BimApi/IIdeaCrossSection.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace IdeaStatiCa.BimApi
+{
+    public interface IIdeaCrossSection : IIdeaObject
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaCrossSection.cs:3]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaCrossSectionByCenterLine.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+using IdeaRS.OpenModel.CrossSection;
+using IdeaRS.OpenModel.Geometry2D;
+
+namespace IdeaStatiCa.BimApi
+{
+    public interface IIdeaCrossSectionByCenterLine : IIdeaCrossSection
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaCrossSectionByCenterLine.cs:6]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaCrossSectionByComponents.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaCrossSectionByComponents : IIdeaCrossSection
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaCrossSectionByComponents.cs:9]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaCrossSectionComponent.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaCrossSectionComponent
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaCrossSectionComponent.cs:9]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaMaterial.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaMaterial : IIdeaObject
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaMaterial.cs:9]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaMaterialConcrete.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+using IdeaRS.OpenModel.Material;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaMaterialConcrete : IIdeaMaterial
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaMaterialConcrete.cs:5]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaMaterialLibrary.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaMaterialLibrary
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaMaterialLibrary.cs:9]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaMaterialReinforcement.cs [hunk 10]
```diff
@@ -0,0 +1,9 @@
+using IdeaRS.OpenModel.Material;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaMaterialReinforcement : IIdeaMaterial
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaMaterialReinforcement.cs:5]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaMaterialSteel.cs [hunk 11]
```diff
@@ -0,0 +1,9 @@
+using IdeaRS.OpenModel.Material;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaMaterialSteel : IIdeaMaterial
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaMaterialSteel.cs:5]: missing comments

File: src/IdeaStatiCa.BimImporter/BimImporter.cs [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+using IdeaStatiCa.BimImporter.ImportedObjects;
+using IdeaStatiCa.BimImporter.Importers;
+using IdeaStatiCa.Plugin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace IdeaStatiCa.BimImporter
+{
+	public class BimImporter : IBimImporter
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimImporter/BimImporter.cs:12]: missing comments

File: src/IdeaStatiCa.BimApi/IIdeaMaterialTendon.cs [hunk 13]
```diff
@@ -0,0 +1,15 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace IdeaStatiCa.BimApi
+{
+	public interface IIdeaMaterialTendon : IIdeaMaterial
```
Comment (mpospisil) [inline src/IdeaStatiCa.BimApi/IIdeaMaterialTendon.cs:9]: missing comments

**Review by mpospisil — APPROVED** [review]

**Pull request merged**